File: 0229-majority-element-ii/0229-majority-element-ii.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        n = len(nums)
        x = n // 3
        mp = {}
        ans = []

        for num in nums:
            mp[num] = mp.get(num, 0) + 1

        for num in mp:
            if mp[num] > x:
                ans.append(num)
        return ans

File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        map<int, int> mp;
        for (auto it : nums) {
            mp[it]++;
        }

        int x = nums.size() / 3;

        vector<int> ans;
        for (auto it : mp) {
            if (it.second > x) {
                ans.push_back(it.first);
            }
        }
        return ans;
    }
};